prow/spyglass/lenses/buildlog/lens.go

```diff
@@ -245,17 +277,107 @@ func (lens Lens) Callback(artifacts []api.Artifact, resourceDir string, data str
 	if !ok {
 		return fmt.Sprintf(missingArtifact, request.Artifact)
 	}
+	if request.Analyze {
+		conf := getConfig(rawConfig)
+		hr, err := analyzeArtifact(artifact, &conf)
+		if err != nil {
+			hr = &highlightResponse{Error: err.Error()}
+		}
+		buf, err := json.Marshal(hr)
+		if err != nil {
+			return err.Error()
+		}
+		return string(buf)
+	}
 	if request.SaveEnd != nil {
 		return storeHighlightedLines(&request, artifact)
 	}
 	return loadLines(&request, artifact, resourceDir, rawConfig)
 }
 
-func storeHighlightedLines(request *lineRequest, artifact api.Artifact) string {
-	err := artifact.UpdateMetadata(map[string]string{
-		focusStart: strconv.Itoa(request.StartLine),
-		focusEnd:   strconv.Itoa(*request.SaveEnd),
+type highlightRequest struct {
```

Comment:
Is this a common/standard interface, or something novel?

Reply:
Novel, planning to work on an implementation next.

Reply:
SGTM!